core/drivers/crypto/caam/acipher/caam_dh.c

```diff
@@ -226,6 +226,14 @@ static TEE_Result do_gen_keypair(struct dh_keypair *key,
 	uint32_t *desc = NULL;
 	uint32_t desclen = 0;
 	int counter = 0;
+	/*
+	 * Set DH CAAM key type to plain text as regression_4007_dh does not
+	 * support black keys.
+	 */
+	enum caam_key_type key_type = CAAM_KEY_PLAIN_TEXT;
+
+	if (key_type == CAAM_KEY_MAX_VALUE)
```

Comment:
Why?

Reply:
Ok so, currently xtest `xtest_tee_test_4007_dh()` does not support CAAM black keys. During DH keypair generation, xtest checks if the returned buffer size is equal to the DH key size. A CAAM black key being bigger than the key itself, one of the check fails.
For xtest compatibility, DH generation is forced to plain text.
Let me know what you think

Reply:
It seems that the test is broken. Is it possible to fix the test? It would be a shame to enforce limits here due to broken tests.